runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -649,12 +818,24 @@ function LanguageTree:included_regions()
     return { {} }
   end
 
-  local regions = {} ---@type Range6[][]
+  local regions = {} ---@type table<integer, Range6[]>
+  local regions_inv = {} ---@type table<integer, (Range6[]|integer)[]>
   for i, _ in pairs(self._trees) do
-    regions[i] = self._trees[i]:included_ranges(true)
+    local region = self._trees[i]:included_ranges(true)
+    prune_empty_ranges(region)
+    if #region > 0 then
+      regions[i] = region
+      regions_inv_insert(regions_inv, i, region)
+    else
+      self._trees[i] = nil
+      if type(self._valid) == 'table' then
```

Comment:
What about the case when `self._valid` is a boolean?

Reply:
If it's boolean, its value doesn't need to be changed.
* if was `true`: the remaining regions are all valid. so still `true`
* if was `false`:
  * if the removed region is the only region in this language tree, any of `false`/`true` is fine
  * otherwise, remaining regions are invalid. so still `false`.